### PR TITLE
Adds integration test annotations, and codenses tests into dropdowns

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/66649
- https://github.com/tgstation/tgstation/pull/70362

## About The Pull Request

- Separated compiling the integration tests and running them as separate steps for organization purposes.
- Added a TEST_ASSERT_NULL(value, reason) and TEST_ASSERT_NOTNULL(value, reason) because those are conceptually simple tests.
- Makes the PASS and FAIL prefixes in the integration test log be green and red for better readability.
- Failure reasons now display the file and line number.
- In order to achieve this, direct calls to Fail() are now wrapped in a macro, TEST_FAIL(), as Fail() itself needs preprocessor stuff passed to it.
- In the midst of updating it, I noticed multiple cases of tests directly calling Fail() and returning when they should have used a better macro, so those were updated. There was at least one case where it appeared that the code assumed that the test ended at Fail(), but made no attempt to do so, such as with the RCD test.
- To take advantage of the previous change, failures are now marked as annotations. Note that outside of github, this creates an ugly-looking line but the primary environment is as a github action.

## Why It's Good For The Game

You can see where the failure happened

image

We dont have to do this nonsense anymore

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Check commit CI tests

You can see prior ones where the tests failed (mostly because I didnt write the checks correctly lol) and the current successful one

I made a separate pr on my branch that showcases the use for it, mainly for when a test has hella lotta fails: 
- https://github.com/Tsar-Salat/BeeStation-Salatland/actions/runs/8492624333/job/23265820702

image

</details>

## Changelog
:cl: rkz. Tastyfish, Mothblocks
server: added annotations and colorized pass/fail to integration tests
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
alrighty. pretty sure I converted all of them correctly. Didnt see any obvious fails so should be g2g.

Lmk if you see any improper checks

**Comment by Tsar-Salat** [comment]
Am I stupid, why is it doing that

**Comment by PowerfulBacon** [comment]
Ideally the CL would also mention mothblocks

File: code/modules/unit_tests/_unit_tests.dm [hunk 1]
```diff
@@ -3,17 +3,26 @@
...
+#define TEST_ASSERT_NULL(a, reason) if (!isnull(a)) { return Fail("Expected null value but received [a]: [reason || "No reason"]", __FILE__, __LINE__) }
 
 /// Asserts that the two parameters passed are equal, fails otherwise
 /// Optionally allows an additional message in the case of a failure
 #define TEST_ASSERT_EQUAL(a, b, message) do { \
 	var/lhs = ##a; \
 	var/rhs = ##b; \
 	if (lhs != rhs) { \
-		return Fail("Expected [isnull(lhs) ? "null" : lhs] to be equal to [isnull(rhs) ? "null" : rhs].[message ? " [message]" : ""]"); \
+		return Fail("Expected [isnull(lhs) ? "null" : lhs] to be equal to [isnull(rhs) ? "null" : rhs].[message ? " [message]" : ""]", __FILE__, __LINE__); \
```
Comment (PowerfulBacon) [inline code/modules/unit_tests/_unit_tests.dm:25]: Does this require 515 to compile?

**Review by PowerfulBacon — APPROVED** [review]
Didn't see this, these are very good and important changes

File: code/modules/unit_tests/_unit_tests.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/unit_tests/_unit_tests.dm:25]: nvm, that's __PROC__

**Review by Crossedfall — APPROVED** [review]
Seems functional

**Pull request merged**